Tests/ToolchainRegistryTests/ToolchainRegistryTests.swift

```diff
@@ -42,339 +42,401 @@ final class ToolchainRegistryTests: XCTestCase {
+      let xcodeDeveloper =
+        tempDir
+        .appendingPathComponent("Xcode.app")
+        .appendingPathComponent("Developer")
+      let toolchains = xcodeDeveloper.appendingPathComponent("Toolchains")
+
+      try makeXCToolchain(
+        identifier: "com.apple.fake.A",
+        opensource: false,
+        path: toolchains.appendingPathComponent("A.xctoolchain"),
+        sourcekitd: true
+      )
+      try makeXCToolchain(
+        identifier: "com.apple.fake.B",
+        opensource: false,
+        path: toolchains.appendingPathComponent("B.xctoolchain"),
+        sourcekitd: true
+      )
 
-    assertEqual(await tr.toolchains.map(\.identifier).sorted(), ["com.apple.fake.A", "com.apple.fake.B"])
+      let tr = ToolchainRegistry(
+        installPath: nil,
+        environmentVariables: [],
+        xcodes: [xcodeDeveloper],
+        libraryDirectories: [],
+        pathEnvironmentVariables: [],
+        darwinToolchainOverride: nil
+      )
+
+      assertEqual(await tr.toolchains.map(\.identifier).sorted(), ["com.apple.fake.A", "com.apple.fake.B"])
+    }
   }
 
   func testIgnoreToolchainsWithWrongExtensions() async throws {
     try SkipUnless.platformIsDarwin("Finding toolchains in Xcode is only supported on macOS")
-    let fs = InMemoryFileSystem()
-    let xcodeDeveloper = try AbsolutePath(validating: "/Applications/Xcode.app/Developer")
-    let toolchains = xcodeDeveloper.appending(components: "Toolchains")
-
-    try makeXCToolchain(
-      identifier: "com.apple.fake.C",
-      opensource: false,
-      path: toolchains.appending(component: "C.wrong_extension"),
-      fs,
-      sourcekitd: true
-    )
-    try makeXCToolchain(
-      identifier: "com.apple.fake.D",
-      opensource: false,
-      path: toolchains.appending(component: "D_no_extension"),
-      fs,
-      sourcekitd: true
-    )
 
-    let tr = ToolchainRegistry(
-      darwinToolchainOverride: nil,
-      fs
-    )
+    try await withTestScratchDir { tempDir in
+      let xcodeDeveloper =
+        tempDir
+        .appendingPathComponent("Xcode.app")
+        .appendingPathComponent("Developer")
+      let toolchains = xcodeDeveloper.appendingPathComponent("Toolchains")
+
+      try makeXCToolchain(
+        identifier: "com.apple.fake.C",
+        opensource: false,
+        path: toolchains.appendingPathComponent("C.wrong_extension"),
+        sourcekitd: true
+      )
+      try makeXCToolchain(
+        identifier: "com.apple.fake.D",
+        opensource: false,
+        path: toolchains.appendingPathComponent("D_no_extension"),
+        sourcekitd: true
+      )
 
-    assertTrue(await tr.toolchains.isEmpty)
+      let tr = ToolchainRegistry(
+        installPath: nil,
+        environmentVariables: [],
+        xcodes: [xcodeDeveloper],
+        libraryDirectories: [],
+        pathEnvironmentVariables: [],
+        darwinToolchainOverride: nil
+      )
 
+      assertEqual(await tr.toolchains.map(\.path), [])
```

Comment:
Isn't `isEmpty` equivalent? Or did you mean to use `compactMap`?

Reply:
It is. But this has the nice side effect of printing which toolchains paths it found when the assertion fails.